### PR TITLE
feat(Guild): add the new nsfw_level property

### PR DESCRIPTION
**Pull request by iShibi**
**Please describe the changes this PR makes and why it should be merged:**

This PR adds the new `nsfw_level` guild property.
Source: [discord-api-docs](https://github.com/discord/discord-api-docs/pull/2976)

**Status and versioning classification:**
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- I know how to update typings and have done so

### CODE REVIEW

**Review by kyranet — CHANGES_REQUESTED** [review]
You should add an enum in the constants, so users don't have to use magic numbers.

**Comment by iShibi** [comment]
I looked at how the already existing enums are being used and as a result of it made some big changes in 606376b. Please let me know if I made any mistake 😺

**Review by kyranet — APPROVED** [review]

File: src/structures/Guild.js [hunk 1]
```diff
@@ -176,6 +169,14 @@ class Guild extends Base {
      */
     this.memberCount = data.member_count || this.memberCount;
 
+    if ('nsfw_level' in data) {
+      /**
+       * The nsfw level of this guild
```
Comment (vaporox) [inline src/structures/Guild.js:174]:
```suggestion
       * The NSFW level of this guild
```

File: typings/index.d.ts [hunk 2]
```diff
@@ -3294,6 +3302,8 @@ declare module 'discord.js' {
     type?: OverwriteType;
   }
 
+  type NsfwLevel = 'DEFAULT' | 'EXPLICIT' | 'SAFE' | 'AGE_RESTRICTED';
```
Comment (vaporox) [inline typings/index.d.ts:3305]:
```suggestion
  type NsfwLevel = keyof typeof NsfwLevels;
```
This should also be moved above `OverwriteData`

**Review by vaporox — CHANGES_REQUESTED** [review]

File: src/structures/Guild.js [hunk 3]
```diff
@@ -176,6 +169,14 @@ class Guild extends Base {
...
+       * @type {NsfwLevel}
+       */
+      this.nsfwLevel = NsfwLevels[data.nsfw_level];
```
Comment (iCrawl) [inline src/structures/Guild.js:177]:
```suggestion
      this.nsfwLevel = NSFWLevels[data.nsfw_level];
```

File: src/util/Constants.js [hunk 4]
```diff
@@ -818,6 +818,16 @@ exports.InteractionResponseTypes = createEnum([
...
 
+/**
+ * NSFW level of a Guild
+ * * DEFAULT
+ * * EXPLICIT
+ * * SAFE
+ * * AGE_RESTRICTED
+ * @typedef {string} NsfwLevel
+ */
+exports.NsfwLevels = createEnum(['DEFAULT', 'EXPLICIT', 'SAFE', 'AGE_RESTRICTED']);
```
Comment (iCrawl) [inline src/util/Constants.js:829]:
```suggestion
exports.NSFWLevels = createEnum(['DEFAULT', 'EXPLICIT', 'SAFE', 'AGE_RESTRICTED']);
```

File: typings/index.d.ts [hunk 5]
```diff
@@ -37,6 +37,13 @@ declare enum InviteTargetType {
   EMBEDDED_APPLICATION = 2,
 }
 
+declare enum NsfwLevels {
```
Comment (iCrawl) [inline typings/index.d.ts:40]:
```suggestion
declare enum NSFWLevels {
```

File: typings/index.d.ts [hunk 6]
```diff
@@ -648,6 +655,7 @@ declare module 'discord.js' {
     ApplicationCommandPermissionTypes: typeof ApplicationCommandPermissionTypes;
     InteractionTypes: typeof InteractionTypes;
     InteractionResponseTypes: typeof InteractionResponseTypes;
+    NsfwLevels: typeof NsfwLevels;
```
Comment (iCrawl) [inline typings/index.d.ts:658]:
```suggestion
    NSFWLevels: typeof NSFWLevels;
```

File: typings/index.d.ts [hunk 7]
```diff
@@ -732,7 +740,7 @@ declare module 'discord.js' {
     public mfaLevel: number;
     public name: string;
     public readonly nameAcronym: string;
-    public nsfw: boolean;
+    public nsfwLevel: NsfwLevel;
```
Comment (iCrawl) [inline typings/index.d.ts:743]:
```suggestion
    public nsfwLevel: NSFWLevel;
```

File: typings/index.d.ts (hunk 2 above)
Comment (iCrawl) [reply typings/index.d.ts:3305]:
```suggestion
  type NSFWLevel = keyof typeof NSFWLevels;
```

**Review by iCrawl — CHANGES_REQUESTED** [review]

File: src/structures/Guild.js [hunk 8]
```diff
@@ -24,6 +24,7 @@ const {
   PartialTypes,
   VerificationLevels,
   ExplicitContentFilterLevels,
+  NsfwLevels,
```
Comment (iCrawl) [inline src/structures/Guild.js:27]:
```suggestion
  NSFWLevels,
```

Comment (iCrawl) [reply src/structures/Guild.js:27]: Woops

**Review by iCrawl — APPROVED** [review]

File: src/structures/Guild.js [hunk 9]
```diff
@@ -176,6 +169,14 @@ class Guild extends Base {
      */
     this.memberCount = data.member_count || this.memberCount;
 
+    if ('nsfw_level' in data) {
+      /**
+       * The NSFW level of this guild
+       * @type {NsfwLevel}
```
Comment (vaporox) [inline src/structures/Guild.js:175]:
```suggestion
       * @type {NSFWLevel}
```

File: src/util/Constants.js [hunk 10]
```diff
@@ -818,6 +818,16 @@ exports.InteractionResponseTypes = createEnum([
   'DEFERRED_CHANNEL_MESSAGE_WITH_SOURCE',
 ]);
 
+/**
+ * NSFW level of a Guild
+ * * DEFAULT
+ * * EXPLICIT
+ * * SAFE
+ * * AGE_RESTRICTED
+ * @typedef {string} NsfwLevel
```
Comment (vaporox) [inline src/util/Constants.js:827]:
```suggestion
 * @typedef {string} NSFWLevel
```

File: typings/index.d.ts [hunk 11]
```diff
@@ -3294,6 +3302,8 @@ declare module 'discord.js' {
     type?: OverwriteType;
   }
 
+  type NSFWLevel = keyof typeof NSFWLevels;
```
Comment (vaporox) [inline typings/index.d.ts:3305]: This should be moved above `OverwriteData`

**Review by vaporox — CHANGES_REQUESTED** [review]

File: src/structures/Guild.js (hunk 8 above)
Comment (iShibi) [reply src/structures/Guild.js]: Yup saw that, just commited the fix. 😄

File: typings/index.d.ts (hunk 11 above)
Comment (iShibi) [reply typings/index.d.ts]: 👍

File: src/util/Constants.js (hunk 10 above)
Comment (iShibi) [reply src/util/Constants.js]: Fixed in the latest commit. I should have waited a little. 😸

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**